packages/nameguard-sdk/src/index.ts

```diff
@@ -489,7 +498,7 @@ class NameGuard {
   }
 
   /**
-   * Inspects a single name with NameGuard. Provides a `NameGuardReport` including:
+   * Inspects a single name with NameGuard. If the `name` is uninspected returns `UninspectedNameGuardReport`; else returns a `NameGuardReport` including:
```

Comment:
@lightwalker-eth Where can I define `UninspectedNameGuardReport`? It is NameGuard report with labels `null` and canonical_name `null`, rating ALERT, highest risk `uninspected`, risk count 1.

Reply:
@djstrong Good question. It's a bit tricky to explain in a comment here.

Here's a simplified example of how to achieve an idea like this in TypeScript. All the below is pseudocode. I might make a typo.

Imagine we have something like the following:

```
interface AbstractMyType {
   value?: string; // NOTE: This is either `undefined` or a `string`.
   otherValue: number;
}
```

We can then also define the following types:
```
interface MyTypeWithValue implements AbstractMyType {
   value: string; // NOTE: This strips away the `undefined` case and always has a `string` value.
   otherValue: number;
}

interface MyTypeWithoutValue implements AbstractMyType {
   value?: undefined; // NOTE: This is ALWAYS undefined
   otherValue: number;
}

type MyType = MyTypeWithValue | MyTypeWithoutValue;

```

Now you can do some cool stuff with this in TypeScript. For example, you can pass in a value of `MyType` into a function. Then inside that function if you check if `value !== undefined` then the type system can know that the code following that you have a `MyTypeWithValue`.

It might be necessary to switch from defining field values as possibly being `null` to using `undefined` instead.

No worries if this is tricky. You're welcome to advance other parts of this PR and assign this step to me if you'd like :+1:

Reply:
@djstrong Also, in the example above `AbstractMyType` never actually needs to be defined, but I find it nice to define that for the purpose of putting code comments on the fields in `AbstractMyType`. That way you can document all the fields in 1 place and not duplicate ideas in comments over and over again on each field of each variation of the abstract type.

Reply:
@lightwalker-eth Thank you. I implemented this:
```ts
export interface InspectedNameGuardReport extends AbstractNameGuardReport {
    labels: LabelGuardReport[];
}

export interface UninspectedNameGuardReport extends AbstractNameGuardReport {
    labels: undefined;
    canonical_name: undefined;
}

export type NameGuardReport = InspectedNameGuardReport | UninspectedNameGuardReport;
```
I don't know which is better here: `extends` or `implements`.

Reply:
@djstrong Sure. I think you've done the right thing here with `extends` :+1:

You should update the definition of `InspectedNameGuardReport` such that it includes the following:

`canonical_name: string;`

Note here how this is not a `?string` but a `string`.

Therefore, with an `AbstractNameGuardReport` we don't know if we'll have `labels` or a `canonical_name`. With an `InspectedNameGuardReport` we're guaranteed to have both (neither can be undefined) and for an `UninspectedNameGuardReport` we're guaranteed to have neither (both must be undefined).